scripts/feature_toggle_report_generator.py

```diff
@@ -0,0 +1,237 @@
+# -*- coding: utf-8 -*-
+
+import datetime
+import io
+import os
+import re
+
+import click
+import jinja2
+import yaml
+
+
+class Ida(object):
+
+    def __init__(self, name):
+        self.name = name
+        self.toggle_states = {}
+        self.annotation_report_path = None
+
+    def add_toggle_data(self, dump_file_path):
+        """
+        Given the path to a file containing the SQL dump for a
+        feature toggle type in an Ida, parse out the information relevant
+        to each toggle and add it to this Ida.
+        """
+        with io.open(dump_file_path, 'r', encoding='utf-8') as dump_file:
+            dump_contents = yaml.safe_load(dump_file.read())
+        for row in dump_contents:
+            toggle_name = row['fields']['name']
+            toggle_type = row['model']
+            toggle_data = row['fields']
+            toggle = ToggleState(toggle_name, toggle_type, toggle_data)
+            if toggle_type in self.toggle_states.keys():
```

Comment:
This logic could be a little simpler if you initialized `self.toggle_states` as a `defaultdict`.

Reply:
Good call.